enterprise/server/cmd/ci_runner/main.go

```diff
@@ -2248,10 +2307,14 @@ func runBashCommand(ctx context.Context, cmd string, env map[string]string, dir
 		return "", &commandError{err, ""}
 	}
 
+	return runCommandWithOutput(ctx, "bash", []string{"-eo", "pipefail", "-c", cmd}, env, dir, outputSink)
+}
+
+func runCommandWithOutput(ctx context.Context, executable string, args []string, env map[string]string, dir string, outputSink io.Writer) (string, *commandError) {
 	var buf bytes.Buffer
 	w := io.MultiWriter(outputSink, &buf)
 
-	if err := runCommand(ctx, "bash", []string{"-eo", "pipefail", "-c", cmd}, env, dir, w); err != nil {
+	if err := runCommand(ctx, executable, args, env, dir, w); err != nil {
```

Comment:
nit: could this refactor be reverted? seems like it's not needed anymore

Reply:
It saves a little duplication betwen `git()` and `runBashCommand` and I could see it being useful in the future, so would prefer to keep it